src/components/App.tsx

```diff
@@ -3552,13 +3459,8 @@ class App extends React.Component<AppProps, AppState> {
         editingLinearElement &&
         editingLinearElement !== this.state.editingLinearElement
       ) {
-        // Since we are reading from previous state which is not possible with
-        // automatic batching in React 18 hence using flush sync to synchronously
-        // update the state. Check https://github.com/excalidraw/excalidraw/pull/5508 for more details.
-        flushSync(() => {
```

Comment:
This `flushSync` was breaking the linearElementEditor tests and removal fixed them, could be breaking though so should be re-tested.

Reply:
this is indeed significant as per the linked issue so we need to keep this until we rewrite our handling. Unclear why it's making the test fail.

@ad1992 any ideas why the test started failing here? (the point position vs position-of-drag does seem like it's on the verge of the threshold, but manually changing it doesn't seem to fix the test — at least it's really flaky).

https://github.com/excalidraw/excalidraw/actions/runs/5814611487/job/15764633529?pr=6759#step:4:1436

```
line.points [
  [ 0, 0 ],
  [ 85.96978489652551, 77.4423262309982 ],
  [ 70, 50 ],
  [ 40, 0 ]
]
{ lastSegmentMidpoint: [ 76.08587175006699, 43.294165939653226 ] }
```